### ISSUE TITLE
Transmution from POD to [u8] (mutable POD reference to &mut [u8] reference)

### ISSUE THREAD

**Issue by frankier** [nabijaczleweli/safe-transmute-rs#17]
Currently it looks like you can only transmute from `[u8]` to some value `T`. Should it not be possible to go the other way, that is from some `T` to `[u8]`? Indeed, it could be desirable to be able to convert between types `T` and `U`, but this would dramatically increase the already large number of functions, so going via `[u8]` makes sense.

As a concrete example of what I want to do (taken from here: https://github.com/frankier/softcomputingforsoftpeople/blob/master/ex1/src/main.rs#L88 ) say I want to do something like with safe_transmute:

```rust
#[derive(Rand, Copy, Clone)]
#[repr(C)]
struct Gene {
    x1: u8,
    x2: u8,
}

fn gene_bits_mut(gene: &mut Gene) -> &mut u16 {
    unsafe { std::mem::transmute::<&mut Gene, &mut u16>(gene) }
}

/// Performs single-point crossover in place, replacing the parents with the children
fn crossover_inplace(mummy: &mut Gene, daddy: &mut Gene, bit: u8) {
    let mummys_bits = gene_bits_mut(mummy);
    let daddys_bits = gene_bits_mut(daddy);
    let mask = (1 << bit) - 1;
    *daddys_bits ^= *mummys_bits & mask;
    *mummys_bits ^= *daddys_bits & mask;
    *daddys_bits ^= *mummys_bits & mask;
}
```

This appears safe as long as Gene is `repr(C)` and POD. Is this within the scope of safe_transmute? Could we make this possible?

**Comment by Enet4** [comment]
IMO this could indeed be within the crate's scope. Whether that could become completely safe is a different story, but since we made `PodTransmutable` an unsafe trait, we delegate that responsibility to the implementer. @nabijaczleweli also has a say on this, but mine is feel free to provide a PR. :)

**Comment by nabijaczleweli** [comment]
(Edited w/marking things as code and adding syntax highlight. Will take a look after sleeping.)

**Comment by nabijaczleweli** [comment]
Released in [`v0.8.0`](https://github.com/nabijaczleweli/safe-transmute-rs/releases/tag/v0.8.0)?

**Comment by frankier** [comment]
Cheers! I didn't even realise Rust was getting C style unions. A safe_union is indeed an interesting prospect.
